### PR TITLE
vscode URI incompatibilities

### LINKED ISSUE #5308: Newest versions of vscode-uri (>=1.0.7) break browser frontend
### Description
 
Theia browser frontend refuses to start after reinstalling `@theia` dependencies.

### Reproduction Steps

After re-building Docker image with `@theia` packages today (5/29/17), my Theia app would no longer start.

Reverting back from `next` to `latest` versions of `@theia` packages, doesn't solve the problem, suggesting that the issue is with a transitive dependency.

Further, browser console reveals the following and only stacktrace:
```js
index.js:73 Failed to start the frontend application.
(anonymous) @ index.js:73
index.js:75 Error: [UriError]: Scheme is missing: {scheme: "", authority: "", path: "", query: "", fragment: ""}
    at index.js:41
    at t.e (index.js:142)
    at new t (index.js:344)
    at Function.e.parse (index.js:254)
    at new e (uri.ts:29)
    at mini-browser-open-handler.ts:63
    at Object.<anonymous> (mini-browser-open-handler.ts:60)
    at n (bootstrap:19)
    at Object.<anonymous> (mini-browser-frontend-module.ts:30)
    at n (bootstrap:19)
```
where `index.js:41` links to `webpack:///./node_modules/vscode-uri/lib/esm/index.js:41`.

Indeed, it appears that 3 more versions (!) of [microsoft/vscode-uri](https://github.com/microsoft/vscode-uri) were released today - `1.0.7`, `1.0.8`, and `2.0.0`.
 
To confirm this culprit (and provide a workaround), I reverted `vscode-uri` to `1.0.6`, using
```json
  "resolutions": {
    "vscode-uri": "<=1.0.6"
  },
```
in `package.json` (using Yarn). After that, the app starts successfully.

**OS and Theia version:**

All Theia packages are using `next` version (which atm is `0.7.0-next.a963e8b5`, from `yarn.lock`).
The Theia app is built in a Docker container based on `node:10-alpine`.
The host is Mac OS 10.13.6, Chrome 74.0.3729.157.

**Diagnostics:**

### PR DESCRIPTION
**Pull request by akosyakov**
fix #5308

### CODE REVIEW

File: packages/core/src/common/uri.ts [hunk 1]
```diff
@@ -14,9 +14,12 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
+import { setUriThrowOnMissingScheme } from 'vscode-uri';
 import Uri from 'vscode-uri';
 import { Path } from './path';
 
+setUriThrowOnMissingScheme(false);
```
Comment (akosyakov) [inline packages/core/src/common/uri.ts]: We have to disable it because of plugin ext: https://github.com/theia-ide/theia/issues/4487

cc @evidolob

**Comment by akosyakov** [comment]
@marcdumais-work it would be good to review it that clients don't need to use yarn resolutions to work around vscode-uri runtime issues.

**Comment by marcdumais-work** [comment]
I am trying to test this using a local NPM registry. I'll update when I have some results.

**Comment by marcdumais-work** [comment]
@akosyakov I think you are on the right track. I built this PR and published the resulting Theia packages in a local registry. Then I used the registry to build an application, re-using the browser example package.json, with packages versions changed to what I published. 

The resulting Theia application starts. I played with it for a minute, and it seems to work ok. Here's the backend log (there was nothing of interest in the frontend log, that I saw): 

```
$ yarn theia start
yarn run v1.15.2
$ /opt/ericsson-dev/theia/examples/browser/b/node_modules/.bin/theia start
root INFO Theia app listening on http://localhost:3000.
root INFO Started watching the git repository: file:///opt/ericsson-dev/theia
root ERROR Uncaught Exception:  Error: Unhandled method getClientId
root ERROR Error: Request 'getClientId' failed
    at Proxy.<anonymous> (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/core/lib/common/messaging/proxy-factory.js:301:33)
    at HostedPluginProcessesCache.<anonymous> (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:85:57)
    at step (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:53:23)
    at Object.next (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:34:53)
    at /opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:28:71
    at new Promise (<anonymous>)
    at __awaiter (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:24:12)
    at HostedPluginProcessesCache.getLazyClientId (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:81:16)
    at HostedPluginProcess.<anonymous> (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-process.js:91:74)
    at step (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-process.js:56:23)
root INFO Using Git [2.21.0] from the PATH. (/usr/bin/git)
root INFO Checking whether '--no-optional-locks' can be used with the current Git executable. Minimum required version is '2.15.0'.
root INFO '--no-optional-locks' is a valid Git option for the current Git version: '2.21.0'.
root INFO [nsfw-watcher: 7680] Started watching: /home/lmcmcds/.theia
root INFO [nsfw-watcher: 7680] Started watching: /opt/ericsson-dev/theia/.theia/settings.json
root INFO [nsfw-watcher: 7680] Started watching: /opt/ericsson-dev/theia/.vscode/launch.json
root INFO [nsfw-watcher: 7680] Started watching: /opt/ericsson-dev/theia/.vscode/settings.json
root INFO [nsfw-watcher: 7680] Started watching: /opt/ericsson-dev/theia/packages/typescript/src/node/typescript-contribution.ts
root INFO [nsfw-watcher: 7680] Started watching: /opt/ericsson-dev/theia
root INFO [nsfw-watcher: 7680] Started watching: /opt/ericsson-dev/theia
root INFO [nsfw-watcher: 7680] Started watching: /opt/ericsson-dev/theia
root INFO Detected keyboard layout from browser API: US (PC)
root WARN Collided keybinding is ignored;  {"command":"outlineView:toggle","keybinding":"ctrlcmd+shift+s"}  collided with  {"command":"file.saveAs","keybinding":"ctrl+shift+s"}
root WARN Could not register keybinding:
  {"command":"outlineView:toggle","keybinding":"ctrlcmd+shift+s"}
Error: "ctrlcmd+shift+s" is in collision with something else [scope:0]
root WARN Collided keybinding is ignored;  {"command":"passthrough","keybinding":"ctrl+k","context":"terminalActive"}  collided with  {"command":"terminal:clear","keybinding":"ctrlcmd+k","context":"terminalActive"}
root WARN Could not register keybinding:
  {"command":"passthrough","keybinding":"ctrl+k","context":"terminalActive"}
Error: "ctrl+k" is in collision with something else [scope:0]
root WARN e.onStart is slow, took: 184.09500003326684 ms
root INFO Using Git [2.21.0] from the PATH. (/usr/bin/git)
root INFO [nsfw-watcher: 7797] Started watching: /home/lmcmcds/.theia
root WARN e.onStart is slow, took: 948.3449999243021 ms
root INFO [nsfw-watcher: 7797] Started watching: /opt/ericsson-dev/theia
root INFO Config file tasks.json does not exist under file:///opt/ericsson-dev/theia
root INFO [nsfw-watcher: 7797] Started watching: /opt/ericsson-dev/theia/.theia/settings.json
root INFO [nsfw-watcher: 7797] Started watching: /opt/ericsson-dev/theia/.vscode/launch.json
root INFO [nsfw-watcher: 7797] Started watching: /opt/ericsson-dev/theia/.vscode/settings.json
root INFO [nsfw-watcher: 7797] Started watching: /opt/ericsson-dev/theia
root INFO [nsfw-watcher: 7797] Started watching: /opt/ericsson-dev/theia
root INFO Stopped watching the git repository: file:///opt/ericsson-dev/theia
root INFO Started watching the git repository: file:///opt/ericsson-dev/theia
root WARN "tslint.enable" preference is not supported
root WARN "search.exclude" preference is not supported
root WARN "lcov.path" preference is not supported
root WARN "lcov.watch" preference is not supported
root WARN "files.insertFinalNewline" preference is not supported
```

**Comment by akosyakov** [comment]
@marcdumais-work thx, i'm trying to fix ci

Can you reproduce this error reliably:
```
root ERROR Uncaught Exception:  Error: Unhandled method getClientId
root ERROR Error: Request 'getClientId' failed
    at Proxy.<anonymous> (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/core/lib/common/messaging/proxy-factory.js:301:33)
    at HostedPluginProcessesCache.<anonymous> (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:85:57)
    at step (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:53:23)
    at Object.next (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:34:53)
    at /opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:28:71
    at new Promise (<anonymous>)
    at __awaiter (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:24:12)
    at HostedPluginProcessesCache.getLazyClientId (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-processes-cache.js:81:16)
    at HostedPluginProcess.<anonymous> (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-process.js:91:74)
    at step (/opt/ericsson-dev/theia/examples/browser/b/node_modules/@theia/plugin-ext/lib/hosted/node/hosted-plugin-process.js:56:23)
```

Please open an issue if so.

**Comment by akosyakov** [comment]
We have to fix #4487 first, otherwise `URI` silently fallback to `file` scheme.

**Comment by akosyakov** [comment]
@svenefftinge i've removed `withoutScheme`, new vscode-uri does not allow URIs without them and silently use `file` scheme. It breaks since developers misused this method to get a path. I've refactored all places to get `path` properly.

File: packages/core/src/common/uri.ts (hunk 1 above)
Comment (svenefftinge) [reply packages/core/src/common/uri.ts]: Please put your comment into the code

Comment (akosyakov) [reply packages/core/src/common/uri.ts]: done

**Comment by akosyakov** [comment]
@marcdumais-work Could you please give a try again and if it is good merge it?

**Comment by marcdumais-work** [comment]
> @marcdumais-work Could you please give a try again and if it is good merge it?

Will do -  have some errands to run; give me a few hours.

**Comment by marcdumais-work** [comment]
@akosyakov 
```
> Can you reproduce this error reliably:

root ERROR Uncaught Exception:  Error: Unhandled method getClientId
[...]
```

I did not see this error with the latest version of this PR

**Comment by marcdumais-work** [comment]
> Could you please give a try again and if it is good merge it?

I re-tested and there are a couple of errors/problems. But I see the same on master. So I'll go ahead and merge this PR, and check if there are existing issues for the problem I saw, and if not create new ones.

**Comment by marcdumais-work** [comment]
@akosyakov Thanks for this fix!

**Pull request merged**